src/pages/HistoryOverviewStatsPage.tsx

```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+import {
+  CategoryScale,
+  Chart as ChartJS,
+  Legend,
+  LinearScale,
+  LineElement,
+  PointElement,
+  Title,
+  TimeScale,
+  Tooltip,
+  ChartOptions,
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+import 'chartjs-adapter-date-fns';
+import { BrewingApi } from '../types/BrewingType';
+import TimeHelper from '../helpers/TimeHelper';
+import { generateDummyBrewingApi } from '../helpers/BrewingStatusLogGenerator';
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  TimeScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Legend
+);
+
+interface Props {
+  selectedBrew: BrewingApi | null;
+}
+
+export const HistoryOverviewStatsPage: React.FC<Props> = ({ selectedBrew }) => {
+  selectedBrew = generateDummyBrewingApi(120);
+  const brewStats = selectedBrew?.StatusLogs;
+
+  const getBrewingDuration = (
+    start: string | undefined,
+    finish: string | undefined
+  ): number => {
+    if (start !== undefined && finish !== undefined) {
+      const startDate = new Date(start);
+      const finishDate = new Date(finish);
+
+      return finishDate.getTime() - startDate.getTime();
+    }
+    return 0;
+  };
+
+  const isDurationMoreThanHour = TimeHelper.isTimeMoreThanHour(
+    getBrewingDuration(selectedBrew?.startedAt, selectedBrew?.finishedAt)
+  );
+
+  const options: ChartOptions<'line'> = {
+    responsive: true,
+    elements: {
+      point: {
+        pointHoverRadius: 8,
+        pointRadius: 7,
+      },
+    },
+    plugins: {
+      legend: {
+        position: 'top' as const,
+        labels: {
+          font: {
+            size: 16,
+          },
+        },
+      },
+      title: {
+        display: false,
+      },
+      tooltip: {
+        titleFont: {
+          size: 20,
+        },
+        bodyFont: {
+          size: 20,
+        },
+        callbacks: {
+          title: function (context) {
+            let value = context[0].label;
+            if (value.length > 5) {
+              const valueSplit = value.split(':');
+              let seconds =
+                +valueSplit[0] * 60 * 60 + +valueSplit[1] * 60 + +valueSplit[2];
+              seconds -= 3600;
+              value = TimeHelper.msToHHMMSS(seconds * 1000);
+              return value;
+            }
+            return value;
+          },
+        },
+      },
+    },
+    scales: {
+      x: {
+        min: isDurationMoreThanHour ? '00:00:00' : '00:00',
+        type: 'time',
+        bounds: 'ticks',
+        ticks: {
+          beginAtZero: true,
+          maxTicksLimit: 20,
+          callback: function (value) {
+            if (typeof value === 'string' && value.length > 5) {
+              const valueSplit = value.split(':');
+              let seconds =
+                +valueSplit[0] * 60 * 60 + +valueSplit[1] * 60 + +valueSplit[2];
+              seconds -= 3600;
+              value = TimeHelper.msToHHMMSS(seconds * 1000);
+              return value;
```

Comment:
som si vsimol ze to pouzivas dva krat, tak to mozes kludne vytiahnut do nejakej funkcie